app.py

```diff
@@ -0,0 +1,75 @@
+from config import Config
+from flask import Flask, render_template, request, flash, redirect, url_for
+from flask_sqlalchemy import SQLAlchemy
+from flask_migrate import Migrate
+from random import choice
+import string
+from datetime import datetime
+from werkzeug import exceptions
+
+
+app = Flask(__name__)
+app.config.from_object(Config)
+
+# database setup
+db = SQLAlchemy(app)
+migrate = Migrate(app, db)
+
+# database model
+class ShortUrls(db.Model):
+    id  = db.Column(db.Integer, primary_key=True)
+    original_url = db.Column(db.String(500), nullable=False)
+    short_id = db.Column(db.String(20), nullable=False, unique=True)
+    created_at = db.Column(db.DateTime(), default=datetime.now(), nullable=False)
+
+# main route
+@app.route('/', methods=['GET', 'POST'])
+def index():
+    if request.method == 'POST':
+        url = request.form['url']
+        short_id = request.form['custom_id']
+
+        if short_id and ShortUrls.query.filter_by(short_id=short_id).first() is not None:
+            flash('Please enter a different custom ID, that one is taken!')
+            return redirect(url_for('index'))
+
+        if not url:
+            flash('The URL is required!')
+            return redirect(url_for('index'))
+
+        if not short_id:
+            short_id = generate_short_id(8)
```

Comment:
Only thing I'd change is check if you have already shortened the given url and just return the corresponding short utl, otherwise you might clock up the db with popular links